session/txn.go

```diff
@@ -510,28 +422,23 @@ func (s *session) HasDirtyContent(tid int64) bool {
 }
 
 // StmtCommit implements the sessionctx.Context interface.
-func (s *session) StmtCommit(memTracker *memory.Tracker) error {
+func (s *session) StmtCommit() error {
 	defer func() {
 		s.txn.cleanup()
 	}()
 
 	st := &s.txn
-	bufLen := st.stmtBufLen()
+	stmtLen := s.txn.countHint()
 
-	if _, err := st.flushStmtBuf(); err != nil {
-		return err
-	}
+	st.flushStmtBuf()
 
-	var err error
 	failpoint.Inject("mockStmtCommitError", func(val failpoint.Value) {
-		if val.(bool) && bufLen > 3 {
-			err = errors.New("mock stmt commit error")
+		if val.(bool) && stmtLen > 3 {
+			err := errors.New("mock stmt commit error")
```

Comment:
Seems the only returned error is failpoint, can we remove the returned error for `StmtCommit`?